rust/macros/module.rs

```diff
@@ -576,23 +587,36 @@ pub(crate) fn module(ts: TokenStream) -> TokenStream {
                 \"#
             );
 
+            /// # Safety
+            /// This function should only be use as a initcall pointer to called by the kernel at load time.
             #[cfg(not(MODULE))]
             #[doc(hidden)]
             #[no_mangle]
-            pub extern \"C\" fn __{name}_init() -> core::ffi::c_int {{
-                __init()
+            pub unsafe extern \"C\" fn __{name}_init() -> core::ffi::c_int {{
+                // SAFETY: This function will only be called by the kernel at load time.
```

Comment:
Similarly, this comment is a bit atypical, but I guess it is fine on its own since it does satisfy the precondition on `__init()`.

Reply:
Done, explained it a bit more